refinery/data_set_manager/search_indexes.py

```diff
@@ -141,14 +143,43 @@ def prepare(self, object):
                FileStoreItem.MultipleObjectsReturned) as e:
             logger.error("Couldn't properly fetch FileStoreItem: %s", e)
             file_store_item = None
-            download_url = "N/A"
+            download_url = NOT_AVAILABLE
         else:
             download_url = file_store_item.get_datafile_url()
             if download_url is None:
-                download_url = (
-                    "N/A" if file_store_item.get_import_status() == SUCCESS
-                    else PENDING
-                )
+                if file_store_item.import_task_id:
+                    logger.debug(
+                        "FileStoreItem with UUID: %s has import_task_id: %s",
+                        file_store_item.uuid,
+                        file_store_item.import_task_id
+                    )
+                    if file_store_item.get_import_status() == SUCCESS:
+                        download_url = NOT_AVAILABLE
+                    else:
+                        # The underlying Celery code in
+                        # FileStoreItem.get_import_status() makes an assumption
+                        # that a result is "probably" PENDING even if it can't
+                        # find an associated Task. See:
+                        # https://github.com/celery/celery/blob/v3.1.20/celery/
+                        # backends/amqp.py#L192-L193 So we double check here to
+                        # make sure said assumption holds up
+                        try:
+                            TaskMeta.objects.get(
+                                task_id=file_store_item.import_task_id
+                            )
+                        except TaskMeta.DoesNotExist:
+                            logger.debug(
+                                "No file_import task for FileStoreItem with "
+                                "UUID: %s",
+                                file_store_item.uuid
+                            )
+                            download_url = NOT_AVAILABLE
+                        else:
+                            download_url = PENDING
+                else:
```

Comment:
If the `else` is much shorter than the `then` consider swapping them: Easier to understand where it's coming from that way.